src/wiseagents/agents/assistant.py

```diff
@@ -0,0 +1,117 @@
+
+import logging
+from threading import Thread
+import threading
+import time
+from typing import Callable, Optional
+import uuid
+
+from wiseagents import WiseAgent, WiseAgentRegistry, WiseAgentTransport
+from wiseagents.wise_agent_messaging import WiseAgentMessage
+import gradio
+
+class AssistantAgent(WiseAgent):
+    """
+    This utility agent start a web interface and pass the user input to another agent.
+    Rhe web interface will be running at http://127.0.0.1:7860
+    """
+    yaml_tag = u'!wiseagents.agents.AssistantAgent'
+    
+    _response_delivery = None
+    _cond = threading.Condition()
+    _response : WiseAgentMessage = None
+    _chat_id = None
+    
+    def __new__(cls, *args, **kwargs):
+        """Create a new instance of the class, setting default values for the optional instance variables."""
+        obj = super().__new__(cls)
+        return obj
+
+    def __init__(self, name: str, description: str , transport: WiseAgentTransport,
+                 destination_agent_name: str):
+        """
+        Initialize the agent.
+
+        Args:
+            name (str): the name of the agent
+            description (str): a description of the agent
+            transport (WiseAgentTransport): the transport to use for communication
+            destination_agent_name (str): the name of the agent to send requests to
+        """
+        self._name = name
+        self._destination_agent_name = destination_agent_name
+        super().__init__(name=name, description=description, transport=transport, llm=None)
+
+    def __repr__(self):
+        """Return a string representation of the agent."""
+        return f"{self.__class__.__name__}(name={self.name}, \
+            description={self.description}, transport={self.transport}, \
+            destination_agent_name={self.destination_agent_name},\
+            response_delivery={self.response_delivery}"
+    
+    def start_agent(self):
+        super().start_agent()
+        self._chat_id= str(uuid.uuid4())
```

Comment:
Just to check, is it intentional that the chat_id is created when starting the agent?

Reply:
Yup, to give the "standard" chatbot experience, keeping memory of what discussed in the chat